nav2_costmap_2d/src/costmap_topic_collision_checker.cpp

```diff
@@ -96,25 +99,27 @@ double CostmapTopicCollisionChecker::scorePose(
 Footprint CostmapTopicCollisionChecker::getFootprint(const geometry_msgs::msg::Pose2D & pose)
```

Comment:
Question: From a software separations of concern standpoint, should be we move this function https://github.com/ros-planning/navigation2/blob/cf10abebfee60fb5cf09df6836b24594b68c083c/nav2_costmap_2d/src/costmap_topic_collision_checker.cpp#L101-L111 to the footprint subscriber itself? It seems like a footprint subscriber's job to take in the footprint and give out a reasonable one given a timestamp of request. There may be users of `FootprintSubscriber` without the topic collision checker that would want a refined & transformed solution out of the box.

Reply:
Yes, this is a good idea. I'm going to put it there along with other cleanup and readability updates of FootprintSubscriber.